### PR TITLE
G-620 | CM Entries | Validate and Update Cm request

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

File: src/main/java/in/projecteka/gateway/registry/RegistryService.java [hunk 1]
```diff
@@ -31,11 +32,34 @@
...
+        return Mono.just(updateCMServiceRequest.build());
+    }
+
+    private Mono<Boolean> validateRequest(CMServiceRequest request) {
+        if (request != null
+                && ((request.getSuffix() == null || request.getSuffix().isBlank())
+                || (request.getUrl() == null || request.getUrl().isBlank())) )
+            return Mono.error(invalidCMRegistryRequest());
+
+        return Mono.just(true);
```
Comment (MounicaSagi) [inline src/main/java/in/projecteka/gateway/registry/RegistryService.java:62]: you can validate this in request body itself by using @valid & @notEmpty/@notNull annotations

Comment (Saisree9) [reply src/main/java/in/projecteka/gateway/registry/RegistryService.java:62]: True. But we want to validate if the URL and suffix with empty string/white spaces also.

Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/registry/RegistryService.java:62]: yeah.. to do that, we have @NotEmpty & @NotBlank annotations.  @NotBlank should be enough to verify null/empty/whitespace characters also.

Comment (Saisree9) [reply src/main/java/in/projecteka/gateway/registry/RegistryService.java:62]: Yeah. But we want to throw Error as a bad request with a valid message. Instead it throws internal server error if the the validation fails. Can we also handle that as well??

**Review by MounicaSagi — APPROVED** [review]

File: src/main/java/in/projecteka/gateway/registry/RegistryService.java [hunk 2]
```diff
@@ -31,11 +32,34 @@
...
+                        ));
+    }
+
+    private Mono<CMServiceRequest> updateCMRequest(CMServiceRequest request) {
+        var updateCMServiceRequest = request.toBuilder();
+
+        if (request.getIsActive() == null)
+            updateCMServiceRequest.isActive(true);
+        if (request.getIsBlocklisted() == null)
+            updateCMServiceRequest.isBlocklisted(false);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/registry/RegistryService.java:51]: these could be property methods.

request.getIsActive() {

return isActive == null ? true : isActive;

}

then you don't need to mutate or need this method at all.

**Pull request merged**